agentai/api.py

```diff
@@ -6,26 +6,31 @@
 import openai
 import requests
 from loguru import logger
-from tenacity import retry, retry_if_exception_type, stop_after_attempt
+from tenacity import retry, retry_if_exception_type, stop_after_attempt, retry_unless_exception_type
 
-from .conversation import Conversation
+from .conversation import Conversation, Message
 from .openai_function import ToolRegistry
 
 logger.disable(__name__)
 
+class InvalidInputError(Exception):
+    pass
 
-@retry(retry=retry_if_exception_type(ValueError), stop=stop_after_attempt(3))
+retry_unless_invalid = retry(retry=retry_unless_exception_type(InvalidInputError), stop=stop_after_attempt(3))
```

Comment:
Don't hard code the `stop_after_attempt` here

Reply:
FIxed.